lib/rules/no-unused-vars.js

```diff
@@ -860,7 +864,563 @@ module.exports = {
+                                            const parentNode = node.parent;
+
+                                            // fix [...[a]] or [...{a}] in function parameters
+                                            if (isFunction(parentNode)) {
+                                                if (parentNode.params.length === 1) {
+                                                    return fixer.removeRange(node.range);
+                                                }
+
+                                                return fixer.removeRange([getPreviousTokenStart(node), node.range[1]]);
+                                            }
+
+                                            // fix rest in nested array pattern like [[a, ...[b]]]
+                                            if (parentNode.type === "ArrayPattern") {
+
+                                                // fix [[...[b]]]
+                                                if (hasSingleElement(parentNode)) {
+                                                    if (parentNode.parent.type === "ArrayPattern") {
+                                                        return fixNestedArrayVariable(parentNode);
+                                                    }
+
+                                                    // fix 'const [...[b]] = foo; and function foo([...[b]]) {}
+                                                    return fixVariables(parentNode);
+                                                }
+
+                                                // fix [[a, ...[b]]]
+                                                return fixer.removeRange([getPreviousTokenStart(node), node.range[1]]);
+                                            }
+
+                                            return null;
+                                        }
+
+                                        // skip fix when variable is reassigned
+                                        if (writeReferences.length > 1 && parentType !== "AssignmentPattern") {
+                                            return null;
+                                        }
+
+                                        // remove declared variables such as var a; or var a, b;
+                                        if (parentType === "VariableDeclarator") {
+                                            if (parent.parent.declarations.length === 1) {
+
+                                                // prevent fix of variable in forOf and forIn loops.
+                                                if (isLoop(parent.parent.parent)) {
+                                                    return null;
+                                                }
+
+                                                // // remove unused declared variable with single declaration like 'var a = b;'
+                                                return fixer.removeRange(parent.parent.range);
```

Comment:
If the code is like:

```js
let a;
(() => a = 1)();
```

The rule now suggests to remove `let a;`, which seems incorrect, since `a` is assigned a value.

Reply:
Done! rule won't give suggestion for this case and even when a function parameter is reassigned.